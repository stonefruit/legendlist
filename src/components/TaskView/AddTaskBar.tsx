// TODO: Add state testing
import { useState } from 'react'
import { PlusIcon } from '@heroicons/react/solid'
import { classNames } from '../../utils'

type Props = {
  addTask(task: { name: string }): void
}
function AddTaskBar({ addTask }: Props) {
  const [name, setName] = useState('')

  // FUNCTIONS

  const validateAndAddTask = () => {
    if (name) {
      addTask({ name })
      setName('')
    }
  }

  const onClickAddTask = () => {
    validateAndAddTask()
  }

  const onChangeContent = (e: React.ChangeEvent<HTMLInputElement>) => {
    setName(e.target.value)
  }

  const onKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      validateAndAddTask()
    }
  }

  return (
    <div className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg">
      <div className="flex-1 px-4 flex justify-between">
        <div className="flex-1 flex">
          <div className="relative w-full text-gray-400 focus-within:text-gray-300">
            <div
              id="add-button"
              onClick={onClickAddTask}
              className={classNames(
                name && 'cursor-pointer',
                'absolute inset-y-0 left-0 flex items-center'
              )}
            >
              <PlusIcon
                className={classNames(
                  name && 'text-yellow-500 hover:text-yellow-600',
                  'h-4 w-4'
                )}
                aria-hidden="true"
              />
            </div>
            <input
              value={name}
              onChange={onChangeContent}
              id="new-task-text-field"
              className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
              placeholder="Add Task. Press Enter to save."
              type="addTask"
              name="addTask"
              onKeyUp={onKeyUp}
            />
          </div>
        </div>
      </div>
    </div>
  )
}

export default AddTaskBar
