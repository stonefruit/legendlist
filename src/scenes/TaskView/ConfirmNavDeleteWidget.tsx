import { CheckIcon, TrashIcon, XIcon } from '@heroicons/react/solid'

type TrashState = 'INACTIVE' | 'ACTIVE' | 'CONFIRM'
type Props = {
  trashState: string
  setTrashState: React.Dispatch<React.SetStateAction<TrashState>>
  onClickDeleteFolder: (id: string) => void
  selectedNavId: string
}
export default function ConfirmNavDeleteWidget({
  trashState,
  setTrashState,
  onClickDeleteFolder,
  selectedNavId,
}: Props) {
  return (
    <div>
      {trashState === 'ACTIVE' && (
        <button
          onClick={() => setTrashState('CONFIRM')}
          className="bg-yellow-100 text-yellow-400 rounded-md p-1 border border-yellow-200 cursor-pointer hover:bg-yellow-50 active:bg-white h-6 w-6 outline-none"
        >
          <TrashIcon className="h-full w-full" />
        </button>
      )}
      {trashState === 'CONFIRM' && (
        <div className="flex justify-center items-center">
          <div className="text-sm">Confirm?</div>
          <button
            onClick={() => {
              setTrashState('ACTIVE')
            }}
            className="ml-1 bg-yellow-100 text-yellow-400 rounded-md p-1 border border-yellow-200 cursor-pointer hover:bg-yellow-50 active:bg-white h-6 w-6 outline-none"
          >
            <XIcon className="h-full w-full" />
          </button>
          <button
            onClick={() => {
              setTrashState('ACTIVE')
              onClickDeleteFolder(selectedNavId)
            }}
            className="ml-1 bg-yellow-100 text-yellow-400 rounded-md p-1 border border-yellow-200 cursor-pointer hover:bg-yellow-50 active:bg-white h-6 w-6 outline-none"
          >
            <CheckIcon className="h-full w-full" />
          </button>
        </div>
      )}
    </div>
  )
}
