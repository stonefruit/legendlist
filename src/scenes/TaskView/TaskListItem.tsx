import {
  ChevronDownIcon,
  ChevronUpIcon,
  TrashIcon,
} from '@heroicons/react/outline'
import { useEffect, useState } from 'react'
import { Task } from '../../types'
import { classNames } from '../../utils'

type Props = {
  task: Task
  updateTask({
    id,
    actualEndDate,
    name,
    folderId,
  }: {
    id: string
    actualEndDate?: number | null
    name?: string
    folderId?: string
  }): Promise<void>
  deleteTask(id: string): Promise<void>
  activeTaskId: string | null
  selectActiveTask(id: string | null): void
  moveTask: (taskId: string, direction: 'UP' | 'DOWN') => Promise<void>
  isTopOfList?: boolean
  isBottomOfList?: boolean
}
export default function TaskListItem({
  task,
  updateTask,
  deleteTask,
  activeTaskId,
  selectActiveTask,
  moveTask,
  isTopOfList,
  isBottomOfList,
}: Props) {
  // Note that name is used so that the react controlled input acts more normally.
  // If async function is used, cursor jumps to end of the word.
  // This still happens if you type very fast.
  const [name, setName] = useState('')
  const isActive = activeTaskId === task.id

  // FUNCTIONS

  const onChangeDone = async (e: React.ChangeEvent<HTMLInputElement>) => {
    e.preventDefault()
    if (e.target.checked) {
      await updateTask({ id: task.id, actualEndDate: Date.now() })
    } else {
      await updateTask({ id: task.id, actualEndDate: null })
    }
  }

  const onClickTask = (id: string) => () => {
    selectActiveTask(id)
  }

  const onChangeName = async (e: React.ChangeEvent<HTMLInputElement>) => {
    setName(e.target.value)
    await updateTask({ id: task.id, name: e.target.value })
  }

  // EFFECTS

  useEffect(() => {
    setName(task.name)
  }, [task.name])

  return (
    <div
      className={classNames(
        isActive ? 'bg-yellow-200' : 'hover:bg-yellow-100',
        'flex px-3 py-3 cursor-pointer border-b group items-center'
      )}
      onClick={onClickTask(task.id)}
    >
      <div
        className={classNames(
          isActive && !task.actualEndDate ? '' : 'invisible',
          'flex flex-col'
        )}
      >
        <ChevronUpIcon
          height={15}
          className={classNames(
            isTopOfList ? 'invisible' : '',
            'text-gray-300 hover:text-gray-800'
          )}
          onClick={() => moveTask(task.id, 'UP')}
        />
        <ChevronDownIcon
          height={15}
          className={classNames(
            isBottomOfList ? 'invisible' : '',
            'text-gray-300 hover:text-gray-800'
          )}
          onClick={() => moveTask(task.id, 'DOWN')}
        />
      </div>
      <div className="flex items-center h-8 pl-3">
        <input
          onClick={(e) => e.stopPropagation()}
          onChange={onChangeDone}
          checked={!!task.actualEndDate}
          id={task.id}
          name="done"
          type="checkbox"
          className="focus:ring-yellow-500 hover:bg-yellow-200 h-4 w-4 text-yellow-600 border-gray-300 rounded cursor-pointer"
        />
      </div>
      <div className={'ml-3 text-sm w-full'}>
        <input
          disabled={!isActive}
          type="text"
          className={classNames(
            isActive
              ? 'bg-yellow-200 group-hover:bg-yellow-200 cursor-text'
              : 'group-hover:bg-yellow-100 cursor-pointer',
            'placeholder-gray-400 pr-6 break-words border-0 bg-yellow-50 p-0 m-0 focus:ring-transparent w-full resize-none outline-none font-medium text-gray-700'
          )}
          placeholder="What would you like to call this item?"
          onChange={onChangeName}
          value={name}
        />
        {/* <p id="comments-description" className="text-gray-400 text-xs">
          {task.orderInFolder}
        </p> */}
      </div>
      <div className="flex flex-row items-center justify-center">
        <TrashIcon
          height={15}
          className={classNames(
            isActive ? '' : 'invisible',
            'text-gray-300 hover:text-gray-800'
          )}
          onClick={() => deleteTask(task.id)}
        />
      </div>
    </div>
  )
}
