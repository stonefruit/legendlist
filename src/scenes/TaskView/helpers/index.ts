export { autoReorderTasks } from './autoReorderTasks'
export { taskSorter } from './utils/taskSorter'
