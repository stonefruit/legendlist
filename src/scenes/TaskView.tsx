import { useEffect, useState } from 'react'
import { NavigationItem, Task, TaskCreateAttributes } from '../types'
import AddTaskBar from './AddTaskBar'
import TaskList from './TaskList.'
import * as db from '../models'

type Props = {
  navigation: NavigationItem[]
  navIndex: number
}
export default function TaskView({ navigation, navIndex }: Props) {
  const [tasks, setTasks] = useState<Task[]>([])
  useEffect(() => {
    const runAsync = async () => {
      const dbTasks = await db.Task.find()
      setTasks(dbTasks)
    }
    runAsync()
  }, [])

  const addTask = async ({ name }: TaskCreateAttributes) => {
    await db.Task.create({
      name,
      content: '',
      priority: 0,
    })
    const dbTasks = await db.Task.find()
    setTasks(dbTasks)
  }

  return (
    <div className="pl-64 flex flex-col flex-1 h-screen">
      <div className="py-6">
        <div className="max-w-7xl mx-auto px-8">
          <h1 className="text-2xl font-semibold text-gray-900">
            {navigation[navIndex]?.name}
          </h1>
        </div>
        <AddTaskBar addTask={addTask} />
        <TaskList tasks={tasks} />
      </div>
    </div>
  )
}
