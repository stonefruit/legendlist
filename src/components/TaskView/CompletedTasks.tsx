// TODO: Add state testing
import { ChevronDownIcon } from '@heroicons/react/outline'
import TaskListItem from './TaskListItem'
import { Task, UpdateTask } from '../../types'

type Props = {
  showCompleted: boolean
  completedTasks: Task[]
  setShowCompleted: React.Dispatch<React.SetStateAction<boolean>>
  updateTask?: UpdateTask
  deleteTask(id: string): void
  selectActiveTask: (id: string | null) => void
  activeTaskId: string | null
}
export default function CompletedTasks({
  showCompleted,
  completedTasks,
  setShowCompleted,
  updateTask,
  deleteTask,
  selectActiveTask,
  activeTaskId,
}: Props) {
  return (
    <div>
      <div className="max-w-7xl mt-4 mx-auto px-8 flex items-center">
        {!showCompleted && completedTasks.length > 0 && (
          <h1
            className="font-semibold text-gray-400 text-sm cursor-pointer hover:text-black"
            onClick={() => setShowCompleted(true)}
          >
            Show Completed&nbsp;({completedTasks.length})
          </h1>
        )}
        {showCompleted && (
          <div
            className="flex items-center justify-center cursor-pointer"
            onClick={() => setShowCompleted(false)}
          >
            <h1 className="text-lg font-semibold text-gray-900">Completed</h1>
            <div className="ml-2 h-5 w-5 flex items-center hover:from-indigo-300">
              {<ChevronDownIcon />}
            </div>
          </div>
        )}
      </div>
      {showCompleted && (
        <div className="py-4">
          {completedTasks.map((task) => {
            return (
              <div key={task.id}>
                <TaskListItem
                  key={task.id}
                  task={task}
                  updateTask={updateTask}
                  deleteTask={deleteTask}
                  selectActiveTask={selectActiveTask}
                  activeTaskId={activeTaskId}
                />
              </div>
            )
          })}
        </div>
      )}
    </div>
  )
}
