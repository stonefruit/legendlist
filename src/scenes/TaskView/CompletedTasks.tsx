import { ChevronDownIcon } from '@heroicons/react/outline'
import { Task } from '../../types'
import TaskListItem from './TaskListItem'

type Props = {
  showCompleted: boolean
  completedTasks: Task[]
  setShowCompleted: React.Dispatch<React.SetStateAction<boolean>>
  updateTask: ({
    id,
    actualEndDate,
    name,
    folderId,
    filePaths,
  }: {
    id: string
    actualEndDate?: number | null | undefined
    name?: string | undefined
    folderId?: string | undefined
    filePaths?: string[] | undefined
  }) => Promise<void>
  deleteTask(id: string): Promise<void>
  selectActiveTask: (id: string | null) => void
  activeTaskId: string | null
  moveTask: (taskId: string, direction: 'UP' | 'DOWN') => Promise<void>
}
export default function CompletedTasks({
  showCompleted,
  completedTasks,
  setShowCompleted,
  updateTask,
  deleteTask,
  selectActiveTask,
  activeTaskId,
  moveTask,
}: Props) {
  return (
    <div>
      <div className="max-w-7xl mt-4 mx-auto px-8 flex items-center">
        {!showCompleted && completedTasks.length > 0 && (
          <h1
            className="font-semibold text-gray-400 text-sm cursor-pointer hover:text-black"
            onClick={() => setShowCompleted(true)}
          >
            Show Completed&nbsp;({completedTasks.length})
          </h1>
        )}
        {showCompleted && (
          <>
            <h1 className="text-lg font-semibold text-gray-900">Completed</h1>
            <div
              className="ml-2 h-5 w-5 flex items-center cursor-pointer hover:from-indigo-300"
              onClick={() => setShowCompleted(false)}
            >
              {<ChevronDownIcon />}
            </div>
          </>
        )}
      </div>
      {showCompleted && (
        <div className="py-4">
          {completedTasks.map((task) => {
            return (
              <div key={task.id}>
                <TaskListItem
                  key={task.id}
                  task={task}
                  updateTask={updateTask}
                  deleteTask={deleteTask}
                  selectActiveTask={selectActiveTask}
                  activeTaskId={activeTaskId}
                  moveTask={moveTask}
                />
              </div>
            )
          })}
        </div>
      )}
    </div>
  )
}
