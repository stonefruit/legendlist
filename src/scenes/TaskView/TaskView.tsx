import { useEffect, useState } from 'react'
import * as models from '../../models'
import { NavigationItem, Task, TaskCreateAttributes } from '../../types'
import NoteView from '../NoteView'
import AddTaskBar from './AddTaskBar'
import ConfirmNavDeleteWidget from './ConfirmNavDeleteWidget'
import TaskListItem from './TaskListItem'
import CompletedTasks from './CompletedTasks'
import { autoReorderTasks, taskSorter } from './helpers'

const ReservedNavIds = { INBOX: 'INBOX', HOME: 'HOME', COMPLETED: 'COMPLETED' }

type TrashState = 'INACTIVE' | 'ACTIVE' | 'CONFIRM'

type Props = {
  navigator: NavigationItem
  selectedNavId: string
  onClickDeleteFolder(id: string): Promise<void>
  navigation: NavigationItem[]
}
export default function TaskView({
  navigator,
  selectedNavId,
  onClickDeleteFolder,
  navigation,
}: Props) {
  const [tasks, setTasks] = useState<Task[]>([])
  const [activeTaskId, setActiveTaskId] = useState<string | null>(null)
  const [trashState, setTrashState] = useState<TrashState>('INACTIVE')
  const [totalTasks, setTotalTasks] = useState(0)
  const [showCompleted, setShowCompleted] = useState(false)
  taskSorter(tasks, 'orderInFolder', 'ASC')
  const activeTask = tasks.find((task) => task.id === activeTaskId) || null

  // FUNCTIONS

  const refreshTasks = async () => {
    const dbTasks = await models.Task.find({ folderId: selectedNavId })
    const updatedTasks = await autoReorderTasks(dbTasks, selectedNavId)

    const wasActiveTaskRemoved =
      updatedTasks.findIndex((task) => task.id === activeTaskId) === -1
    if (wasActiveTaskRemoved) {
      setActiveTaskId(null)
    }
    setTasks(updatedTasks)
  }

  const addTask = async ({ name }: TaskCreateAttributes) => {
    const newTaskId = await models.Task.create({
      name,
      folderId: selectedNavId,
      orderInFolder: 0,
      content: [],
      priority: 0,
      actualEndDate: null,
      actualStartDate: null,
      plannedEndDate: null,
      plannedStartDate: null,
      filePaths: [],
    })
    await refreshTasks()
    setActiveTaskId(newTaskId)
  }

  const reorderTask = async (taskId: string, beforeTaskId: string | null) => {
    const dbTasks = await models.Task.find({ folderId: selectedNavId })
    taskSorter(dbTasks, 'orderInFolder', 'ASC')

    const currentTaskIndex = dbTasks.findIndex((task) => task.id === taskId)
    const beforeTaskIndex = dbTasks.findIndex(
      (task) => task.id === beforeTaskId
    )
    let newTaskIndex: number
    if (beforeTaskIndex > -1) {
      dbTasks.splice(beforeTaskIndex, 0, dbTasks[currentTaskIndex])
      newTaskIndex = beforeTaskIndex
    } else {
      dbTasks.splice(dbTasks.length, 0, dbTasks[currentTaskIndex])
      newTaskIndex = dbTasks.length
    }
    const taskToRemoveIndex = dbTasks.findIndex(
      (task, index) => task.id === taskId && newTaskIndex !== index
    )
    dbTasks.splice(taskToRemoveIndex, 1)

    const tasksToUpdate = dbTasks.map((task, index) => {
      task.orderInFolder = index
      return task
    })
    await models.Task.bulkPut(tasksToUpdate)
    await refreshTasks()
  }

  const moveTask = async (taskId: string, direction: 'UP' | 'DOWN') => {
    const currentTaskIndex = tasks.findIndex((task) => task.id === taskId)
    if (direction === 'UP' && currentTaskIndex !== 0) {
      await reorderTask(taskId, tasks[currentTaskIndex - 1].id)
    }
    if (direction === 'DOWN' && currentTaskIndex !== tasks.length - 1) {
      const isTaskMovingToLast = currentTaskIndex + 1 === tasks.length - 1
      if (isTaskMovingToLast) {
        await reorderTask(taskId, null)
      } else {
        await reorderTask(taskId, tasks[currentTaskIndex + 2].id)
      }
    }
  }

  const updateTask = async ({
    id,
    actualEndDate,
    name,
    folderId,
    filePaths,
  }: {
    id: string
    actualEndDate?: number | null
    name?: string
    folderId?: string
    filePaths?: string[]
  }) => {
    // Bring task to top when it is moved to another folder
    const isMovingFolder = !!folderId
    const isSettingToUncompleted = actualEndDate === null
    const shouldBringTaskToTop = isMovingFolder || isSettingToUncompleted
    const orderInFolder = shouldBringTaskToTop ? -1 : undefined
    const taskMovedFromElsewhere = shouldBringTaskToTop || actualEndDate

    await models.Task.update({
      id,
      actualEndDate,
      name,
      folderId,
      orderInFolder,
      filePaths,
    })
    const updatedTask = await models.Task.get({ id })
    if (taskMovedFromElsewhere) {
      setActiveTaskId(null)
    } else if (updatedTask) {
      setActiveTaskId(updatedTask.id)
    }
    await refreshTasks()
  }

  const selectActiveTask = (id: string | null) => {
    setActiveTaskId(id)
  }

  const completedTasks = tasks.filter((task) => task.actualEndDate)
  const uncompletedTasks = tasks.filter((task) => !task.actualEndDate)
  taskSorter(completedTasks, 'actualEndDate', 'DSC')

  // EFFECTS

  useEffect(() => {
    // Reset state
    setActiveTaskId(null)
    setShowCompleted(false)
    refreshTasks()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [selectedNavId])

  useEffect(() => {
    const runAsync = async () => {
      await refreshTasks()
    }
    runAsync()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  useEffect(() => {
    if (tasks.length !== totalTasks) {
      setTotalTasks(tasks.length)
    }
  }, [tasks, totalTasks])

  useEffect(() => {
    if (totalTasks > 0) {
      setTrashState('INACTIVE')
    } else {
      setTrashState('ACTIVE')
    }
  }, [totalTasks])

  useEffect(() => {
    const runAsync = async () => {
      if (activeTaskId === null) {
        setActiveTaskId(null)
        return
      }
      const updatedTask = await models.Task.get({ id: activeTaskId })
      if (!updatedTask) {
        setActiveTaskId(null)
        return
      }
      setActiveTaskId(updatedTask.id)
    }
    runAsync()
  }, [activeTaskId])

  useEffect(() => {
    if (completedTasks.length === 0) {
      setShowCompleted(false)
    }
  }, [completedTasks])

  useEffect(() => {
    if (!showCompleted) {
      setActiveTaskId(null)
    }
  }, [showCompleted])

  return (
    <div className="ml-64 flex flex-row bg-yellow-50 opacity-90">
      <div className="flex flex-col flex-1 h-screen overflow-y-auto">
        <div className="py-6">
          <div className="max-w-7xl mx-auto px-8 flex justify-between items-center">
            <h1 className="text-2xl font-semibold text-gray-900">
              {navigator.name}
            </h1>
            {!(ReservedNavIds as any)[navigator.id] && (
              <ConfirmNavDeleteWidget
                trashState={trashState}
                setTrashState={setTrashState}
                onClickDeleteFolder={onClickDeleteFolder}
                selectedNavId={selectedNavId}
              />
            )}
          </div>
          <AddTaskBar addTask={addTask} />

          <div className="mx-auto">
            <div className="py-4">
              {uncompletedTasks.map((task) => {
                return (
                  <div key={task.id}>
                    <TaskListItem
                      key={task.id}
                      task={task}
                      updateTask={updateTask}
                      selectActiveTask={selectActiveTask}
                      activeTaskId={activeTaskId}
                      moveTask={moveTask}
                    />
                  </div>
                )
              })}
            </div>
            <CompletedTasks
              showCompleted={showCompleted}
              completedTasks={completedTasks}
              setShowCompleted={setShowCompleted}
              updateTask={updateTask}
              selectActiveTask={selectActiveTask}
              activeTaskId={activeTaskId}
              moveTask={moveTask}
            />
          </div>
        </div>
      </div>
      <NoteView
        task={activeTask}
        navigation={navigation}
        updateTask={updateTask}
      />
    </div>
  )
}
