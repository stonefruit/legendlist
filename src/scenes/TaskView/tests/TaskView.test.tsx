import TestRenderer from 'react-test-renderer'
import TaskView from '../TaskView'

const { act } = TestRenderer

const navigation = [
  {
    id: 'INBOX',
    name: 'Inbox',
  },
  {
    id: 'f939e3fe-e063-4070-a9d6-57cdee7f3c92',
    name: 'New Folder',
  },
]

const navigator = {
  id: 'INBOX',
  name: 'Inbox',
}

test('TaskView', async () => {
  const component = TestRenderer.create(
    <TaskView
      navigator={navigator}
      selectedNavId="INBOX"
      onClickArchiveFolder={async () => {}}
      navigation={navigation}
    />
  )
  const testInstance = component.root
  let tree = component.toJSON() as any
  expect(tree).toMatchSnapshot()

  // const addButton = testInstance.findByProps({ id: 'add-button' })
  // await addButton.props.onClick()
  // tree = component.toJSON()
  // expect(tree).toMatchSnapshot()
})
