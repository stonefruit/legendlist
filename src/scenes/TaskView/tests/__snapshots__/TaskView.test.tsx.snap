// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView 1`] = `
<div
  className="ml-64 flex flex-row bg-yellow-50 opacity-90"
>
  <div
    className="flex flex-col flex-1 h-screen overflow-y-auto"
  >
    <div
      className="py-6"
    >
      <div
        className="max-w-7xl mx-auto px-8 flex justify-between items-center"
      >
        <h1
          className="text-2xl font-semibold text-gray-900"
        >
          Inbox
        </h1>
      </div>
      <div
        className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg"
      >
        <div
          className="flex-1 px-4 flex justify-between"
        >
          <div
            className="flex-1 flex"
          >
            <div
              className="relative w-full text-gray-400 focus-within:text-gray-300"
            >
              <div
                className="absolute inset-y-0 left-0 flex items-center"
                id="add-button"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="h-4 w-4"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
              <input
                className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
                id="new-task-text-field"
                name="addTask"
                onChange={[Function]}
                onKeyUp={[Function]}
                placeholder="Add Task. Press Enter to save."
                type="addTask"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="mx-auto"
      >
        <div
          className="py-4"
        />
        <div>
          <div
            className="max-w-7xl mt-4 mx-auto px-8 flex items-center"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col flex-1 w-64 border-l h-screen justify-center overflow-y-hidden"
  >
    <div
      className="flex justify-center"
    >
      Task Notes will appear here
    </div>
  </div>
</div>
`;
